backends/bmv2/simple_switch/simpleSwitch.cpp

```diff
@@ -810,7 +810,25 @@ Util::IJson* ExternConverter_log_msg::convertExternFunction(
     auto str = ctxt->conv->convert(mc->arguments->at(0)->expression);
     params->append(str);
     if (mc->arguments->size() == 2) {
-        auto val = ctxt->conv->convert(mc->arguments->at(1)->expression);
+        auto arg1 = mc->arguments->at(1)->expression;
+        // this must be a list expression, with all components
+        // evaluating to integral types.
+        auto argType = ctxt->typeMap->getType(arg1);
+        if (auto ts = argType->to<IR::Type_List>()) {
+            for (auto tf : ts->components) {
+                if (!tf->is<IR::Type_Bits>()) {
```

Comment:
does it still work with SerEnum arguments?

Reply:
Line 134 from issue2201-bmv2.p4 tests for this case

Reply:
Since the testdata/p4_16_samples/issue2201-bmv2.p4 is marked as XFAIL, I don't think anything is really tested as far as these backend changes are concerned? But if `tf->is<IR::Type_Bits>()` is true for SerEnum types, that works for me.

Reply:
I added a new test 2201-1-bmv2.p4 which tests this case.